qiskit_ionq/helpers.py

```diff
@@ -566,6 +569,54 @@ def get_n_qubits(backend: str, _fallback=100) -> int:
         return _fallback
 
 
+def retry(
+    exceptions=Exception,
+    tries=-1,
+    delay=0,
+    max_delay=None,
+    backoff=1,
+    jitter=0,
+):  # pylint: disable=too-many-positional-arguments
+    """Retry decorator with exponential backoff.
+
+    Args:
+        exceptions (Exception or tuple): the exception(s) to check. may be a tuple of
+            exceptions to check.
+        tries (int): number of times to try (not retry) before giving up. -1 means infinite tries.
+        delay (float): initial delay between retries in seconds.
+        max_delay (float): the maximum value of delay in seconds. default is no max.
+        backoff (float): backoff multiplier e.g. value of 2 will double the delay each retry.
+        jitter (float or callable): extra seconds added to delay between retries to
+            prevent thundering herd effect. may be fixed value or a callable that
+            returns a value.
+    """
+
+    def deco_retry(func):
```

Comment:
If we're going to write our own, can we add tests? Separately, log statements on failures would also be useful. 

https://pypi.org/project/reretry/ came up as another alternative as well

Reply:
added a test and logging :)